### PR TITLE
Start a page with examples and recipes

### PR DESCRIPTION
**Pull request by tronical**
This is the beginning of a page to collect common things users need to
do, in a format that they can copy & paste into their application code.

There's a lot more that could be done though.

### CODE REVIEW

**Comment by ogoffart** [comment]
> I wonder how to detect in js if we are running in rustdoc or Sphinx? Or alternatively: how could we instruct an injected script to pick Rust or C++ and remove the other?

We do this kind of stuff for the highlighting already:
https://github.com/slint-ui/slint/blob/b259656f4c9abceefb05592d4eaf54127c2c4aa0/docs/resources/slint-docs-highlight.html#L55

**Comment by ogoffart** [comment]
I think maybe early on an example to use callback in a global would be great

**Comment by tronical** [comment]
> I think maybe early on an example to use callback in a global would be great

Done

**Comment by tronical** [comment]
This depends on #1094 but it still has two TODOs (fix crash in previews - might require implementing full GL context loss/recovery handling, and rewrite default size and resize handling)

File: docs/recipes/recipes.md [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+    }
+}
+```
+
+<details data-snippet-language="rust">
+<summary>Rust code</summary>
+In Rust you can write
+
+```rust,no_run
+slint::slint!(import { Recipe } from "docs/recipes/button_native.slint";);
```
Comment (lukas-jung) [inline docs/recipes/recipes.md:44]: if it is not run, the directory doesn't need to be valid. should probably just be `"button_native.slint"`.

Comment (ogoffart) [reply docs/recipes/recipes.md:44]: it is not previewed by the wasm in the browser.  But it is still tested by the doctest test that checks for slint compile error.

**Comment by adoble** [comment]
An example on how to code a screen flow, i.e. how can I switch from one screen to another. 

The print demo does something like this, but it took me quite some time to see how it is done.

**Comment by tronical** [comment]
That's an excellent suggestion

**Pull request merged**